includes/wizards/class-performance-wizard.php

```diff
@@ -0,0 +1,170 @@
+
+/**
+ * Easy interface for setting up general store info.
+ */
+class Performance_Wizard extends Wizard {
+	/**
+	 * The slug of this wizard.
+	 *
+	 * @var string
+	 */
+	protected $slug = 'newspack-performance-wizard';
+
+	/**
+	 * The capability required to access this wizard.
+	 *
+	 * @var string
+	 */
+	protected $capability = 'install_plugins';
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		parent::__construct();
+		add_action( 'rest_api_init', [ $this, 'register_api_endpoints' ] );
+	}
+	/**
+	 * Get the name for this wizard.
+	 *
+	 * @return string The wizard name.
+	 */
+	public function get_name() {
+		return esc_html__( 'Performance', 'newspack' );
+	}
+
+	/**
+	 * Get the description of this wizard.
+	 *
+	 * @return string The wizard description.
+	 */
+	public function get_description() {
+		return esc_html__( 'Set up and maintain Progressive Web App functionality.', 'newspack' );
+	}
+
+	/**
+	 * Get the expected duration of this wizard.
+	 *
+	 * @return string The wizard length.
+	 */
+	public function get_length() {
+		return esc_html__( '10 minutes', 'newspack' );
+	}
+
+	/**
+	 * Register the endpoints needed for the wizard screens.
+	 */
+	public function register_api_endpoints() {
+		register_rest_route(
+			'newspack/v1/wizard/',
+			'/performance/',
+			[
+				'methods'             => \WP_REST_Server::READABLE,
+				'callback'            => [ $this, 'api_get_settings' ],
+				'permission_callback' => [ $this, 'api_permissions_check' ],
+			]
+		);
+		register_rest_route(
+			'newspack/v1/wizard/',
+			'/performance/',
+			[
+				'methods'             => \WP_REST_Server::CREATABLE,
+				'callback'            => [ $this, 'api_update_settings' ],
+				'permission_callback' => [ $this, 'api_permissions_check' ],
+			]
+		);
+	}
+
+	/**
+	 * Handle GET endpoint.
+	 */
+	public function api_get_settings() {
+		return rest_ensure_response( $this->get_settings() );
+	}
+
+	/**
+	 * Handle POST endpoint.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 */
+	public function api_update_settings( $request ) {
+		if ( empty( $request['settings'] ) ) {
+			return new WP_Error( 'newspack_invalid_update', 'Invalid update' );
+		}
+		$configuration_manager = Configuration_Managers::configuration_manager_class_for_plugin_slug( 'progressive-wp' );
+		if ( is_wp_error( $configuration_manager ) ) {
+			return $configuration_manager;
+		}
+		$settings = $request['settings'];
+		if ( ! empty( $settings['site_icon'] ) ) {
+			$configuration_manager->update( 'site_icon', $settings['site_icon'] );
+		}
+		if ( ! empty( $settings['push_notification_server_key'] ) ) {
+			$configuration_manager->update( 'firebase-serverkey', $settings['push_notification_server_key'] );
+		}
+		if ( ! empty( $settings['push_notification_sender_id'] ) ) {
+			$configuration_manager->update( 'firebase-senderid', $settings['push_notification_sender_id'] );
+		}
```

Comment:
We also need to update [the `pwp_firebase_credentials_set` option](https://github.com/SayHelloGmbH/progressive-wordpress/blob/aab1bd0a3a193dd602dc785070cfea7255abc43b/inc/funcs.php#L37) to `'yes'` if the creds are set, otherwise it won't show up as having push notification set (the credentials do successfully show up in the settings if I follow the push notifs card here, though):
![localhost_wp-admin_admin php_page=progressive-wordpress](https://user-images.githubusercontent.com/7317227/60468083-55d45f00-9c0d-11e9-8155-6f060fcde875.png)

Reply:
I added the logic to set it to `yes` in https://github.com/Automattic/newspack-plugin/pull/141/commits/33f8b830831efe02bc747ecc40543422df8b4089. All of the toggles aren't linked to Progressive WP fields, but they should be. When we figure out this logic and hook them up we can also build in a way to set `pwp_firebase_credentials_set` to `no`.